test/mlj_interface.jl

```diff
@@ -42,15 +42,19 @@
 
     @testset "generic interface tests" begin
         @testset "CatBoostRegressor" begin
-            failures, summary = MLJTestInterface.test([CatBoostRegressor],
-                                                      MLJTestInterface.make_regression()...;
+            data = MLJTestInterface.make_regression()
+            X = DataFrame(data[1])
```

Comment:
You shouldn't need to convert to a `DataFrame` if you make the genericity changes explained above. Your interface must work for any table (with `Tables.istable(X) = true`).